streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -283,7 +283,11 @@ public synchronized V putIfAbsent(K key, V value) {
     private void restoreAllInternal(Collection<KeyValue<byte[], byte[]>> records) {
         try (WriteBatch batch = new WriteBatch()) {
             for (KeyValue<byte[], byte[]> record : records) {
-                batch.put(record.key, record.value);
+                if (record.value == null) {
+                    batch.remove(record.key);
```

Comment:
I believe it to be a delete operation to the underlying store.  I've been reading through the source to confirm.